src/useRequest/useRequest.tsx

```diff
@@ -54,10 +54,9 @@ export function useRequest<T>(options?: Options): RequestResponse<T> {
       const { data: response } = await fn;
 
       if (params && params.concat) {
-        const concatResponse = _concat(params.concat, data, response);
-        setData(concatResponse);
+        setData((prevState: Options) => _concat(params.concat!, prevState, response));
```

Comment:
You should return concat response.

I would add this line of code:
`
let newData;
setData((prevState) => { 
newData = _concat(params.concat!, prevState, response);
return newData
})

return newData;
`